js/src/setup-ads/ads-stepper/index.js

```diff
@@ -12,24 +13,43 @@ import SetupAccounts from './setup-accounts';
 import AdsCampaign from '.~/components/paid-ads/ads-campaign';
 import SetupBilling from './setup-billing';
 
-const AdsStepper = ( props ) => {
-	const { formProps } = props;
+/**
+ * @param {Object} props React props
+ * @param {Object} props.formProps Form props forwarded from `Form` component.
+ * @fires gla_setup_ads with `{ triggered_by: 'step1-continue-button' | 'step2-continue-button' , action: 'go-to-step2' | 'go-to-step3' }`.
+ * @fires gla_setup_ads with `{ triggered_by: 'stepper-button', action: 'go-to-step1' | 'go-to-step2' }`.
+ */
+const AdsStepper = ( { formProps } ) => {
 	const [ step, setStep ] = useState( '1' );
 
 	// Allow the users to go backward only, not forward.
 	// Users can only go forward by clicking on the Continue button.
 	const handleStepClick = ( value ) => {
 		if ( value < step ) {
+			recordEvent( 'gla_setup_ads', {
+				triggered_by: 'stepper-button',
```

Comment:
Since we are adding `step1-continue-button `, `step2-continue-button `...etc I wonder if it would be better to add the origin step here. i.e `step1-stepper-button`

Reply:
Tweaked in https://github.com/woocommerce/google-listings-and-ads/pull/2111/commits/6ed73ac53ae0935ad0b3182f8113ea63770c972e.

I guess which one would be better will depend on the users of tracking data, the goal here is to reduce the confusion of the data itself.

The `stepN-continue-button` naming means the continue button in step N. But the stepper buttons exist in all steps, so I named it `stepper-stepN-button`.